sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3239,7 +3239,7 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
       messageParameters = Map(
         "protobufType" -> protobufType,
         "toType" -> toSQLType(sqlType)),
-      cause = Option(cause.getCause))
+      cause = Option(cause))
```

Comment:
@rangadi there are 5 changes in this file. Shall we have tests to verify all of them?

Reply:
I am not sure if that is required. There is no reason to pass `ex.getCause()` rather than `ex`. It was simple oversight. I don't think we need tests for that.

Reply:
ok